### DISCUSSION TITLE
MC -> Paste: Full Page Modal

### DISCUSSION THREAD

**Discussion by ceyerkes** [twilio-labs/paste#2812] [General] [answered]
Hello! I'm in the process of gathering more info on [composable components](https://airtable.com/invite/l?inviteId=invDOGIPlN9uZWwo8&inviteToken=3ee06d5f079231f9e7883528a2e3be5ac188a138f1bf0936c3c20733aafd6b2b&utm_medium=email&utm_source=product_team&utm_content=transactional-alerts) that exist in MC but don't currently exist in Paste. 

Full page modals are used extensively in MC in editing flows, as well as in the design library ([screenshots](https://www.figma.com/file/EubaymZfxmFAfa3ksfkKxS/Paste-Migration-2022%2F23-%5BWIP%5D?node-id=373%3A260118)). Instead of proposing a Paste-ified version of this component, I wanted to check in with the team to see if there are any other recommendations for this type of pattern.

**Comment by SiTaggart** [comment, marked as answer]
👋🏼 Hi!

Yeah, this is a bit of a tricky one really, because we really struggle to justify it when we look at the reasoning behind the need.

Most folks tend to give one main reason for a full page modal, which is keeping context with the action you were just performing or where you came from. The problem with that is in two areas:

1. If it's modal you can't access or use the information that you require context from due to the nature of modal dialogs. It renders the content beneath inert and out of reach.
2. If it's full screen you can't _see_ the content you require context from

So really all you need is a quick way to go _back_ to the screen you were just on, and that often just feels like a page navigation problem, rather than a component problem.

So we've, up until now, really only suggested making it a page. A normal, everyday page, that you navigate to like any other, and give them a prominent, easy to fine navigation action that takes them back to _exactly where_ they were.

Are there any other reasons why that wouldn't work for you in this context?

**Reply by ceyerkes** [reply]
Perfect - thanks @SiTaggart! I'll work on creating full screen modal flows as separate pages, and reach out if we run into any reasons why we would need to maintain the full screen modal status quo.

**Reply by ceyerkes** [reply]
Hey @SiTaggart - how does Paste feel about a collapsible nav? The more i dig into the editor designs, the clearer it is that we need some sort of solution to create more real estate in the editing space. Another idea is opening the editor in a new tab like Canva does. 

Here's a rough example of how I think it could work:

![Screen Shot 2022-12-02 at 9 56 54 AM](https://user-images.githubusercontent.com/87141551/205344855-549d40ef-94bf-4404-befe-c8ccee5869e0.png)

**Reply by SiTaggart** [reply]
Yeah, totally fine. We do it in Console and Flex already for these kind of reasons

**Reply by SiTaggart** [reply]
As an aside, these look GREAT!!

**Reply by ceyerkes** [reply]
Oh amazing, and thank you!

Do those patterns exist in Paste, or should I reach out to designers on those teams to find out more about how they implemented a collapsible nav? I'm wondering how to handle the sub pages when the nav is collapsed:

![Screen Shot 2022-12-02 at 10 23 12 AM](https://user-images.githubusercontent.com/87141551/205349721-21d2f340-e836-4841-8ebd-d59c1430a6d9.png)

**Reply by SiTaggart** [reply]
We don't at the moment. 2nd half of next year we will once our unified navigation components come out, but for now sync with the console and flex folks. They both do it slightly differently right now, but it should give you a starting point to get going.

**Reply by ceyerkes** [reply]
The only example i've been able to grab from console is this one sans nav icons - i was initially thinking that we'd use the nav icons, but it looks like the one in console removes icons and adds an expand icon at the bottom:

![Screen Shot 2022-12-05 at 10 49 29 AM](https://user-images.githubusercontent.com/87141551/205706879-163609ef-f282-4ab9-a6fb-6a82bac25e30.png)

Would love your feedback on icons vs. no icons when you have a sec.

**Reply by ceyerkes** [reply]
Reopening this discussion on full page modals in the context of an [image editor](https://www.figma.com/file/AThoZNMoCXViDddWDbKb0d/Content-Editor?node-id=481%3A210265).

I'm currently working through 3 image editor designs:

**Wide Modal** (current Paste)

![Screenshot 2023-01-31 at 9 00 49 AM](https://user-images.githubusercontent.com/87141551/215812816-f45dc124-6f00-4f89-bbb1-65cc4a7e3e54.png)


**Extra Wide Modal**

![Screenshot 2023-01-31 at 9 01 06 AM](https://user-images.githubusercontent.com/87141551/215812924-aa582961-0aa2-43b9-a070-702cabe87e2f.png)


**Full Screen Modal** 

![Screenshot 2023-01-31 at 9 01 20 AM](https://user-images.githubusercontent.com/87141551/215813007-845cf54a-0982-416b-a956-ef694094b6ea.png)

I completely understand avoiding full screen modals for the 2 reasons @SiTaggart mentioned above, but I think there's a case for making an exception for a focused editor flow. The team is leaning towards Extra wide and full screen, but would appreciate any feedback on these when you have a sec, thanks!

cc: @serifluous

**Reply by gloriliale** [reply]
@ceyerkes Was there a particular reason for not wanting to use the first option? I'm mostly thinking of how it works well for wide images like the one in these examples, which the other options cut off. Also it allows you to see the same amount of filtered examples.

**Reply by ceyerkes** [reply]
@gloriliale we could definitely change the dimensions of the preview to be wider in the 2nd and 3rd, this is just a quick mockup - there's plenty of room on either side. The first example gives us significantly less preview room (350px square), which is why my PM and I like the 2nd and 3rd options better (600+px square).

![Screenshot 2023-01-31 at 12 40 16 PM](https://user-images.githubusercontent.com/87141551/215865171-df5ea446-0b58-44a7-a70b-d1d0812579ed.png)

**Reply by ceyerkes** [reply]
Side-by-side with image sizes increased on full and extra wide modals:

![Screenshot 2023-01-31 at 12 47 17 PM](https://user-images.githubusercontent.com/87141551/215866594-32af6be6-41fe-49e9-ba96-db144030d101.png)

**Reply by serifluous** [reply]
Rounding out the convo here: @kelseycavitt, @ceyerkes, and I met, and we landed on a solution to use the existing Content Editor/UI page and swap out the Email preview for an image preview/editing—so no modal needed.

Closing the discussion again for now.

**Comment by ceyerkes** [comment]
Reopening this convo - @cfree has concerns around not being able to influence the actual URL routing within the application (he can explain this better) and is recommending the big modal over the full page:

![Screenshot 2023-02-10 at 10 49 43 AM](https://user-images.githubusercontent.com/87141551/218161037-aac73cef-8cf3-475f-8e04-10591db01413.png)

**Reply by kelseycavitt** [reply]
@ceyerkes Hi, can you elaborate? Is he wanting a dedicated "page" so there is a URL? 

Also, can @cfree ?
